### ISSUE TITLE
Index trait not implemented for BString

### ISSUE THREAD

**Issue by eddybruel** [BurntSushi/bstr#190]
When trying to use bstr in my own project, I noticed that slicing a BString yields a &[u8], while slicing a BStr yields a &BStr (which is what I'd expect).

Upon closer inspection, it looks like the Index trait is implemented for BStr, but not for BString, which is not what I'd expect. I'm not sure if this was a deliberate design decision or just an oversight, so I just wanted to bring this to your attention.

**Comment by BurntSushi** [comment]
Yup. This looks like an oversight. Slicing a `BString` I guess should probably yield a `BStr`. This can't be fixed without a breaking change, which I don't plan on releasing any time soon, if ever.

You can probably work around this by derefing a `BString` to `BStr` and then slicing.

**Comment by eddybruel** [comment]
Thanks for confirming. I can simply work around it by wrapping the resulting slice in a call to BStr::new(..). Bit of a wart, but nothing too bad. 

Appreciate the quick reply!

**Comment by eddybruel** [comment]
Closing this since it cannot be fixed without a breaking change, which doesn't seem worth it.

**Comment by BurntSushi** [comment]
I'll keep it open in the event that a `bstr 2.0` ever occurs.
